cluster/src/main/java/com/hartwig/pipeline/resource/TargetRegionsFiles.java

```diff
@@ -0,0 +1,41 @@
+package com.hartwig.pipeline.resource;
+
+import java.util.Collections;
+import java.util.List;
+
+import com.google.api.client.util.Lists;
+import com.hartwig.pipeline.Arguments;
+import com.hartwig.pipeline.CommonArguments;
+import com.hartwig.pipeline.execution.vm.BashCommand;
+import com.hartwig.pipeline.execution.vm.VmDirectories;
+import com.hartwig.pipeline.execution.vm.unix.MkDirCommand;
+
+public class TargetRegionsFiles
+{
+    private static final String TARGETED_REGIONS_DIR = String.format("%s/%s", VmDirectories.RESOURCES, ResourceNames.TARGET_REGIONS);
+
+    private static final String OVERRIDE_TARGETED_REGIONS_DIR = String.format("%s/override", TARGETED_REGIONS_DIR);
+
+    public static List<BashCommand> overrides(final Arguments arguments) {
+
+        List<BashCommand> commands = Lists.newArrayList();
+
+        if(arguments.targetRegionsDir().isEmpty())
+            return commands;
+
+        commands.add(new MkDirCommand(OVERRIDE_TARGETED_REGIONS_DIR));
+
+        String targetRegionsDir = arguments.targetRegionsDir().get();
+        commands.add(() -> String.format("gsutil -qm cp -n %s/* %s", targetRegionsDir, OVERRIDE_TARGETED_REGIONS_DIR));
+        return commands;
+    }
+
+    public static void parseArguments(final ResourceFiles resourceFiles, final CommonArguments arguments) {
+        if (arguments.useTargetRegions()) {
+            resourceFiles.setTargetRegionsDir(TARGETED_REGIONS_DIR);
```

Comment:
As mentioned we try not to mutate incoming parameters, but also we try to void mutability in general especially on the interfaces abstracting access to data. Now clients have to know something about the order of things, ie they could call getTargetRegionsDir before the setter is called and assume we're not in target mode. This is why I first implemented with a decorator on an immutable resource files. 

I'm ok to leave for now, as its more about future proofing. But will likely come back to this before next release.